pkg/kubernetes/client/get.go

```diff
@@ -49,15 +54,19 @@ func (k Kubectl) GetByState(data manifest.List) (manifest.List, error) {
 }
 
 type getOpts struct {
-	allNamespaces bool
-	stdin         string
+	allNamespaces  bool
+	ignoreNotFound bool
+	stdin          string
 }
 
 func (k Kubectl) get(namespace, kind string, selector []string, opts getOpts) (manifest.Manifest, error) {
 	// build cli flags and args
 	argv := []string{
 		"-o", "json",
 	}
+	if opts.ignoreNotFound {
+		argv = append(argv, "--ignore-not-found")
```

Comment:
This creates a problem, as the following logic does not handle the case where kubectl returns nothing on `stdout`, causing the subsequent `json.Unmarshal` to fail. We should find a way to signal consumers that no resources were found to be returned, for example by introducing a new `ErrorNothingReturned`

Reply:
I have introduced a new `ErrorNothingReturned`. The error is propagated up to the `uid()` function where it is handled and empty map of uids is returned from `uids()` function. Let me know if you think that the error should be propagated and handled further up the call stack.

Reply:
I'm not sure if it's correct that I don't pass stderr produced by kubectl along with the `ErrorNothingReturned`. My reasoning is that if the exit code was 0 then it's not necessary to handle stderr because it should be empty.